### PR TITLE
fix remaining ICDS reports tests in python 3

### PR DESCRIPTION
**Pull request by nickpell**

### CODE REVIEW

File: custom/icds_reports/reports/prevalence_of_undernutrition.py [hunk 1]
```diff
@@ -328,7 +333,10 @@ def get_prevalence_of_undernutrition_sector_data(domain, config, loc_level, loca
         if sql_location.name not in result_set:
             chart_data['blue'].append([sql_location.name, 0])
 
-    chart_data['blue'] = sorted(chart_data['blue'])
+    chart_data['blue'] = sorted(
+        chart_data['blue'],
+        key=lambda loc_and_value: (loc_and_value[0] is not None, loc_and_value)
```
Comment (emord) [inline custom/icds_reports/reports/prevalence_of_undernutrition.py]: @calellowitz This feels like an error to me for sql_location to not have a name. Any ideas if this is just in test data and won't be hit in prod?

**Comment by nickpell** [comment]
✅

File: custom/icds_reports/tests/test_disha_dump.py [hunk 2]
```diff
@@ -5,13 +5,15 @@
...
 from io import open
+
+from django.test import TestCase
+
 from corehq.util.files import TransientTempfile
 from custom.icds_reports.reports.disha import DishaDump
 
 
-class DishaFileBuildTest(SimpleTestCase):
+class DishaFileBuildTest(TestCase):
```
Comment (calellowitz) [inline custom/icds_reports/tests/test_disha_dump.py]: why does this need to change?

File: custom/icds_reports/reports/prevalence_of_undernutrition.py (hunk 1 above)
Comment (calellowitz) [reply custom/icds_reports/reports/prevalence_of_undernutrition.py]: does this even happen in test data? but yeah agree that locations should always have names. I would have guessed that was enforced at the db level, although it appears not to be. How does this change affect the sorting?

File: custom/icds_reports/reports/prevalence_of_undernutrition.py [hunk 3]
```diff
@@ -208,10 +208,15 @@ def get_prevalence_of_undernutrition_data_chart(domain, config, loc_level, show_
...
-    )
+    all_locations = [
+        {
+            'loc_name': key,
+            'percent': value
+        }
+        for key, value in six.iteritems(best_worst)
+    ]
+    all_locations_sorted_by_name = sorted(all_locations, key=lambda x: x['loc_name'])
+    all_locations_sorted_by_percent_and_name = sorted(all_locations_sorted_by_name, key=lambda x: x['percent'])
```
Comment (calellowitz) [inline custom/icds_reports/reports/prevalence_of_undernutrition.py]: is there anyway to do this without sorting twice? it doesnt seem like the first value is used at all.

**Review by calellowitz — COMMENTED** [review]
couple small questions but seems good

File: custom/icds_reports/tests/test_disha_dump.py (hunk 2 above)
Comment (nickpell) [reply custom/icds_reports/tests/test_disha_dump.py]: a couch doc is accessed during this test.  For some unknown reason it wasn't an issue in python 2.

File: custom/icds_reports/reports/prevalence_of_undernutrition.py (hunk 3 above)
Comment (nickpell) [reply custom/icds_reports/reports/prevalence_of_undernutrition.py]: The goal is to specify a deterministic sort order (by name) when the percents are equal.  This is required for tests python 2/3 changes the default dictionary iteration order.

```all_locations_sorted_by_name``` is used as an input to the second sort that creates ```all_locations_sorted_by_percent_and_name```.

re: double sort - I'm going to do a second pass on all these changes (and those in https://github.com/dimagi/commcare-hq/pull/24116/) to implement this recommendation where possible: https://github.com/dimagi/commcare-hq/pull/24116/files/f7741d55e7ad87de96d093f6db2cc9de8a415cef#diff-7231be0c95659b29a34b182f3aa8e8b5

File: custom/icds_reports/reports/prevalence_of_undernutrition.py (hunk 1 above)
Comment (nickpell) [reply custom/icds_reports/reports/prevalence_of_undernutrition.py]: Python 3 does not permit sorting iterables containing strings or numbers when a value of ```None``` is present.

File: custom/icds_reports/reports/prevalence_of_undernutrition.py (hunk 3 above)
Comment (calellowitz) [reply custom/icds_reports/reports/prevalence_of_undernutrition.py]: :+1:

File: custom/icds_reports/tests/test_disha_dump.py (hunk 2 above)
Comment (calellowitz) [reply custom/icds_reports/tests/test_disha_dump.py]: :+1:

**Pull request merged**